provisioner/ibm-s3fs-provisioner.go

```diff
@@ -399,7 +441,65 @@ func (p *IBMS3fsProvisioner) Provision(options controller.VolumeOptions) (*v1.Pe
 		}
 	}
 
+	contextLogger.Info("ConfigBucketAccessPolicy: " + strconv.FormatBool(*ConfigBucketAccessPolicy) + ", SetAccessPolicy: " + pvc.SetAccessPolicy)
+
+	//add check for region = BNNP
+	if ConfigBucketAccessPolicy != nil && *ConfigBucketAccessPolicy && pvc.SetAccessPolicy != "false" {
+		grpcSess = p.GRPCBackend.NewGrpcSession()
+		cc := &grpcClient.GrpcSes{}
+		conn, err := grpcSess.GrpcDial(cc, *SockEndpoint, grpc.WithInsecure(), grpc.WithBlock(), grpc.WithDialer(UnixConnect))
+		if err != nil {
+			return nil, fmt.Errorf(pvcName+":"+clusterID+":failed to establish grpc-client connection: %v", err)
+		}
+
+		providerClient = p.IBMProvider.NewIBMProviderClient(conn)
+		if providerClient != nil && conn != nil {
```

Comment:
Shouldn't we call `defer conn.Close()` when `conn != nil` ? Do we really need to check for `providerClient`?
As conn is already open so irrespective of value of `providerClient`, we should call `defer conn.Close()`..right?

Reply:
yes. we need not check for `providerClient`. changed accordingly.